### PR TITLE
Liberty

### PR DESCRIPTION
**Pull request by Noahliberty**
Hey Guys,

Moved all head, meta, and header code into one file header.ejs and included on all pages. Mongo DB up and running. Changed routing pattern to restful for continuity, adjusted pathing of static files so everything shows up!

Let me know what you think.

Thanks

### CODE REVIEW

**Comment by Noahliberty** [comment]
adding latest route and page

File: app.js [hunk 1]
```diff
@@ -3,27 +3,108 @@ const express    = require("express"),
...
+const reviewSchema = new mongoose.Schema({
+	title: String,
+	description: String,
+	imageUrl: String,
+	author: String,
+	authorUrl: String,
+	courseUrl: String,
+	review: String,
+	price: Number,
+	isFree: Boolean,
```
Comment (vansi42) [inline app.js]: Do we need this field? We can use price as 0

File: app.js [hunk 2]
```diff
@@ -3,27 +3,108 @@ const express    = require("express"),
...
 app.set("view engine", "ejs");	
 
 // Serve static files
 var path = require('path');
 app.use(express.static(path.join(__dirname, 'static')));			  
-		  
-// Route for main page
+
+// 		  Mongo Schema for new reviews, eventually break off into seperate folder to req
+const reviewSchema = new mongoose.Schema({
```
Comment (vansi42) [inline app.js]: Do we need an ID field? for the detailed course page? How do we select the course to view?

File: app.js [hunk 3]
```diff
@@ -3,27 +3,108 @@ const express    = require("express"),
...
+	const imageUrl = req.body.imageUrl;
+// 	Save as new var object
+	const newReview = {title: title, author: author, authorUrl: authorUrl, reviewTitle: reviewTitle, reviewDetails: reviewDetails, price: price, isFree: isFree, courseUrl: courseUrl, imageUrl: imageUrl, description: description };
+// 	Add to data base
+	Review.create(newReview, (err, newlyCreated) => {
+		if(err) {
+			console.log(err);
+		} else {
+			console.log(newlyCreated);
+			res.redirect("/")
```
Comment (vansi42) [inline app.js]: Redirect to /courses will save us a redirect

File: app.js [hunk 4]
```diff
@@ -3,27 +3,108 @@ const express    = require("express"),
...
+	const price = req.body.price;
+	const isFree = req.body.isFree;
+	const courseUrl = req.body.courseUrl;	
+	const imageUrl = req.body.imageUrl;
+// 	Save as new var object
+	const newReview = {title: title, author: author, authorUrl: authorUrl, reviewTitle: reviewTitle, reviewDetails: reviewDetails, price: price, isFree: isFree, courseUrl: courseUrl, imageUrl: imageUrl, description: description };
+// 	Add to data base
+	Review.create(newReview, (err, newlyCreated) => {
+		if(err) {
+			console.log(err);
```
Comment (vansi42) [inline app.js:101]: handle errors: maybe we should have an error page we can redirect to in this case

File: app.js [hunk 5]
```diff
@@ -3,27 +3,108 @@ const express    = require("express"),
...
+		if (err) {
+			console.log(err)
+		} else {
+			res.render("landing",{reviews:allReviews});
+		}
+	});
+	
+	// New - Route to show form for new courses
+app.get("/courses/new", (req, res) => {
+		const tags = [{id: 1, title: "CSS"},
```
Comment (vansi42) [inline app.js:63]: todo later: get tags list from DB

File: app.js (hunk 1 above)
Comment (Noahliberty) [reply app.js]: The isFree field? Sure and add an if (price === 0) {
price = free}
or something?

File: app.js (hunk 2 above)
Comment (vansi42) [inline app.js:15]: Do we need an ID field? for the detailed course page?

File: app.js (hunk 1 above)
Comment (vansi42) [inline app.js:24]: Do we need this field? We can use price as 0

File: app.js (hunk 3 above)
Comment (vansi42) [inline app.js:104]: Redirect to /courses will save us a redirect

**Review by vansi42 — COMMENTED** [review]
Looks great! Please check comments

**Comment by Noahliberty** [comment]
Hi,

In response.

handle errors: maybe we should have an error page we can redirect to in this case - Absolutely. this was a plan for the future and easy to change.

todo later: get tags list from DB - Yiup can add a collection ;later and fine tune

Do we need an ID field? for the detailed course page? - Yes and luck Mongo creates a Unique ID for each entry we can then look up to display the right Course and later review and associate courses and reviews or comments with users

Do we need this field? We can use price as 0 - Sure have a if(price === 0) { price = free} (or whatever)

Redirect to /courses will save us a redirect - true good point

**Pull request merged**